1_fetch/src/get_nwis_data.R

```diff
@@ -1,37 +1,34 @@
-create_dirs <- function(site_num){
-  download_files_to <- file.path('1_fetch/out', paste0('nwis_', site_num, '_data.csv'))
-  return(download_files_to)
-}
 
-download_nwis_site_data <- function(site_num, download_files_to, parameterCd = '00010', startDate="2014-05-01", endDate="2015-05-01"){
+download_nwis_site_data <- function(site_num, download_files_to){
+  #create a directory to download the files to
+  download_files_to <- file.path('1_fetch/out', paste0('nwis_', site_num, '_data.csv'))
   
   # readNWISdata is from the dataRetrieval package
   data_out <- readNWISdata(sites=site_num, service="iv", 
-                           parameterCd = parameterCd, startDate = startDate, endDate = endDate)
+                           parameterCd = '00010', startDate="2014-05-01", endDate="2015-05-01")
   
   # -- simulating a failure-prone web-sevice here, do not edit --
   set.seed(Sys.time())
   if (sample(c(T,F,F,F), 1)){
     stop(site_num, ' has failed due to connection timeout. Try tar_make() again')
   }
   # -- end of do-not-edit block
-  correct_dir <- download_files_to[grep(site_num,download_files_to)]
-  write_csv(data_out, file = correct_dir)
-  return(correct_dir)
+  write_csv(data_out, file = download_files_to)
+  return(download_files_to)
 }
 
-concat_files_to_df <- function(){
-  downloaded_data <- list.files(path = '1_fetch/out', pattern = '_data.csv', full.names = T)
+concat_files_to_df <- function(csv_1, csv_2, csv_3, csv_4, csv_5){
+  downloaded_files <- c(csv_1, csv_2, csv_3, csv_4, csv_5)
   data_out <- data.frame()
```

Comment:
This should work fine but a little cleaner implementation would be have 'download_files' as the sole input argument to the function and just pass in the file names as a vector rather than individually. Then you can delete the 'downloaded_files <- c(csv_1, csv_2, csv_3, csv_4, csv_5)' line.

Reply:
I left this as is because I wasn't sure where to create the ```downloaded_files``` object in order to keep the inputs to ```concat_files_to_df()``` the individual site targets

Reply:
Ok to do that - what I was suggesting was if you wrap your individual site targets in your 'site_data_concat' target in '_targets.R' with 'c( )' then your input argument to the 'concat_files_to_df()' can be the 'download_files' variable directly and your function doesn't then have to have a predefined number of input files.

Reply:
Got it! Implemented in the next commit